cmake/projects/OpenSSL/externals/nasm_path_configure.cmake.in

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.0)
+
+message("################################")
+set(ENV{PATH} "@NASM_ROOT@/bin;$ENV{PATH}")
+message(STATUS "$ENV{PATH}")
+
+execute_process(COMMAND
+    perl Configure "${opt}" ${shared} "--prefix=@HUNTER_PACKAGE_INSTALL_PREFIX@" "--openssldir=${openssl_dir}")
```

Comment:
You should check for exit code, see example:
* https://github.com/ruslo/hunter/blob/v0.23.50/cmake/projects/Qt/ep-stages/qt-configure.cmake.in#L365-L369

Reply:
Variables `opt`, `shared` and `openssl_dir` are not defined in this scope. You should use `@opt@`, `@shared@` and `@openssl_dir@`.

Reply:
Check that `@HUNTER_PACKAGE_INSTALL_PREFIX@` defined too:
```cmake
if("@HUNTER_PACKAGE_INSTALL_PREFIX@" STREQUAL "")
  message(FATAL_ERROR "HUNTER_PACKAGE_INSTALL_PREFIX is empty")
endif()
```

Reply:
Yes `@HUNTER_PACKAGE_INSTALL_PREFIX@` is empty

Reply:
Set some variable in OpenSSL build scheme and use the variable in script:

```cmake
# cmake/projects/OpenSSL/schemes/url_sha1_openssl_windows_1_1_plus.cmake.in

set(openssl_install_dir "@HUNTER_PACKAGE_INSTALL_PREFIX@")
```

```cmake
# nasm_path_configure.cmake.in

if("@openssl_install_dir@" STREQUAL "")
  # error message
endif()

execute_process(COMMAND perl Configure "--prefix=@openssl_install_dir@" ...)
```

Reply:
well set in build scheme but not in nasm_path_configure.cmake